### ISSUE TITLE
List of recent editors + screen reader should not read input box placeholder

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#94587]
More details here https://github.com/microsoft/vscode/issues/94360#issuecomment-610136168

fyi @bpasero

**Comment by isidorn** [comment]
@jvesouza insiders with this fix should be out end of week. So try it out around that time and let us know if the issue got fixed. Thanks!

**Comment by jvesouza** [comment]
@isidorn  I just installed code-insiders version 1.45.0, commit 2225d77536d12fea2eda04930f431a5eb0930bba, and the problem still exists.

When I press ctrl+tab orca reads type the name of an editor to open it.

**Comment by isidorn** [comment]
I just tried this out and orca does not read "type the name of an editor to open it". For me the following is read "edt active".
@jvesouza do you see the same with latest?

Anyways since this is probably a potential workaround I will leave this issue open and pushiing it out of the April milestone.

**Comment by jvesouza** [comment]
@isidorn 
Orca announces the following:
'Type the name of an editor to open it.'
'editable combo box'
'edt active '
'opens menu'

**Comment by isidorn** [comment]
I tried this again with latest VS Code insider and I could not reproduce. However what I hit was the following:
1. VS Code insiders, orca, have some files open in VS Code
2. Ctrl + P, type `edt `
3. The list of active editors is shown. Use up / down to navigate between them

Orca properly reads aria label of each editor, however it always reads the aria label of the first editor in the list as well.
@joanmarie can you maybe look into this, since it seems to work just fine with NVDA?

**Comment by joanmarie** [comment]
@jvesouza: Are you pressing and releasing both Ctrl+Tab repeatedly? Or are you pressing Ctrl+Tab, keeping Ctrl held down, and then tabbing repeatedly? If you always press and release both Ctrl and Tab, then focus keeps switching between the editable combo box and the editor.  But if you keep Ctrl held down and press and release only Tab, focus remains in the list and, for me, Orca doesn't repeat all the verbosity you describe.

@isidorn: I cannot reproduce what you describe. What version of Orca are you using?

**Comment by jvesouza** [comment]
@joanmarie  I hold down the control key while pressing and release the tab key.
Would it help if I sent an orca debug file?

**Comment by joanmarie** [comment]
@jvesouza Sure.

**Comment by joanmarie** [comment]
So looking at your debug.out from Orca, it seems to confirm what I was asking. Here's what I see:

You press Ctrl+Tab and Orca announces "Type the name of an editor to open it. Editable combo box. edt active. opens menu." The reason why is:
* "Type the name of an editor to open it" is the accessible name of the combo box which just gained focus. That comes from VSCode.
* "Editable combo box" is the role. The role comes from VSCode/Chromium. Orca adds on "editable" so you know you can type in it.
* "edt active" is the displayed value of the combo box/input.
* "opens menu" is because the aria-haspopup has a value of "menu"edt active

According to your debug.out, as you continue to press and release only Tab, Orca does not say "Type the name of an editor to open it." However, I do see Orca say that again a couple of times. What happened is that you released Ctrl. That caused focus to go back to the editor. So when you pressed Ctrl+Tab again, focus left the editor and you were put back into the combo box.

So.... What is the bug to be fixed here?

Mind you, I do see some chattiness that it might make sense to eliminate (e.g. when you move into the listbox from the combobox, and also the repetition of the results count). But I don't understand what is being reported here. Please clarify. Thanks!

**Comment by jvesouza** [comment]
The problem is when I press ctrl + tab just once. In this case, orca does not advertise the editor I'm switching to.
Assuming that I am editing 3 open files, file1, file2 and file3, and that I am editing file1.
If I press ctrl + tab, release the tab key and keep the ctrl key pressed, orca does not announce the name of the next editor to be focused on.
If I have a lot of editors open and the desired editor is the next to the list, this can make the process of finding the desired file in the list of editors a little bit slower as I need to press the  ctrl + tab a few times until I hear the editor's name wanted.

I'm sorry if the explanation is not clear, my English is not very good.

**Comment by joanmarie** [comment]
Oh.... Hmmm. When you press and release Ctrl+Tab, the stuff I described in my previous comment is still taking place. Let me see if I can detect that condition in Orca and filter out some of the associated noise. Then I'll go from there. Thanks for the clarification. And you're English is great -- and way better than my non-existent Portuguese. :grin:

**Comment by joanmarie** [comment]
@isidorn: Clearing the cache accessibility state of the combobox which claims focus (i.e. in order to see if it already gave up focus) is not solving the problem. Is there any way you could add a slight delay so that quickly pressing and releasing both ctrl and tab does not cause the quick pick list to temporarily gain focus?

**Comment by isidorn** [comment]
@joanmarie sorry I forgot to update. I just updated to latest Orca version and now I am on 3.37.91pre. With that version I can still reproduce what I mentioned [above](https://github.com/microsoft/vscode/issues/94587#issuecomment-673977634)

@chmarti for the quick pick focus question in the [last comment](https://github.com/microsoft/vscode/issues/94587#issuecomment-674805855).
@joanmarie I believe the focus is all the time in the quick pick input, and that the list always has "fake focus" via `activedescendant`. Do you see focus jumping around when you do the `Ctrl+Tab` sequence?

**Comment by joanmarie** [comment]
@isidorn When the value of `aria-activedescendant` changes, user agents typically emit accessible focus-change events to cause the screen reader to present the new location.

Can you please tell me specifically what Orca is repeating each time you press and release tab (not just the first time), with control held down?

**Comment by isidorn** [comment]
@joanmarie I am not holding down control. My steps are the following:
1. VS Code insiders, orca, have some files open in VS Code
2. Ctrl + P, type `edt ` - note that there is a space at the end!
3. The list of active editors is shown. Use up / down to navigate between them

Let's say the first file in the list is `first.txt`, second is `second.txt` etc... As I move focus down the list and the focus item is for example `fifth.txt`. Orca reads the following:
`first.txt Editor Group One, fifth.txt Editor Group One`
So for some items it appends the aria label of the first item in the list (even though it is not focused)

**Comment by chrmarti** [comment]
@isidorn You are right, it is using `aria-activedescendant`. The focus shouldn't go to the list unless the user navigates to it using Tab.
